ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -14101,6 +14107,110 @@ void SemanticErrorChecker::checkEmbedActivity(IHqlExpression * call)
     }
 }
 
+void SemanticErrorChecker::checkGrouping(IHqlExpression * expr)
+{
+    IHqlExpression * attrs = expr->queryChild(3);
+    if (attrs && queryAttributeInList(groupedAtom, attrs))
```

Comment:
I think this test is wrong (and was wrong in the original).  It should be 

    expr->hasAttribute(groupedAtom)

see ecl/regress/aggds4.ecl for an example which changes.

Reply:
Actually, it was probably correct in the original - the difference is at this point the expression has been created and the no_comma lists have been expanded.